core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/recipe/RecipeEngine.java

```diff
@@ -56,16 +53,16 @@ public void executePreClusterManagerRecipes(Stack stack, Collection<Recipe> reci
         }
     }
 
-    // note: executed when LDAP config is present, because later the LDAP sync is hooked for this salt state in the top.sls.
     public void executePostAmbariStartRecipes(Stack stack, Collection<Recipe> recipes) throws CloudbreakException {
-        if ((stack.getCluster() != null && ldapConfigService.isLdapConfigExistsForEnvironment(stack.getEnvironmentCrn(), stack.getName()))
-                || recipesFound(recipes, POST_CLOUDERA_MANAGER_START)) {
+        if (shouldExecuteRecipeOnStack(recipes, POST_CLOUDERA_MANAGER_START)) {
             orchestratorRecipeExecutor.postClusterManagerStartRecipes(stack);
         }
     }
 
-    public void executePostInstallRecipes(Stack stack) throws CloudbreakException {
-        orchestratorRecipeExecutor.postClusterInstall(stack);
+    public void executePostInstallRecipes(Stack stack, Collection<Recipe> recipes) throws CloudbreakException {
+        if (shouldExecuteRecipeOnStack(recipes, POST_CLUSTER_INSTALL)) {
```

Comment:
if you skip this step, this script won't run: https://github.com/hortonworks/cloudbreak/blob/master/orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh
from here: https://github.com/hortonworks/cloudbreak/blob/master/orchestrator-salt/src/main/resources/salt/salt/recipes/post-cluster-install.sls#L32-L46